### DISCUSSION TITLE
Cold start to getServerSideProps is very slow

### DISCUSSION THREAD

**Discussion by davidwitten** [vercel/community#427] [Help] [answered]
I have a very simple getServerSideProps function (made simply for testing):

```
export const getServerSideProps = async (ctx) => {
    return {
        props: {test: null},
    };
```

However, when I deploy my website through Vercel and try loading a page that uses this it first takes ~5 seconds. Once the serverless function is warm, it's almost instantaneous.

Is there any way to resolve this? I would be willing to upgrade my plan to Pro if that would fix it.

**Comment by kachkaev** [comment]
👋 @davidwitten have you found a solution or a workaround? Facing the same problem in vercel.com deployments. Pages that use `getServerSideProps` takes 5-10 seconds to ‘warm up’.

**Comment by Pieparker** [comment, marked as answer]
Hey y'all! Have you [checked this out](https://github.com/vercel/community/discussions/307#discussioncomment-2152407)?

**Reply by kachkaev** [reply]
👋 @Pieparker! Yep, I have seen that comment. Unfortunately I haven’t been able to bring the initial load down to 1-2 seconds even if made my `getServerSideProps` a noop. Cold boot time was still ~10 seconds for our project ([blockprotocol.org](https://blockprotocol.org)).

Here is a solution I came up with:

- [/.github/workflows/warm-up-vercel.yml](https://github.com/blockprotocol/blockprotocol/blob/f009d8145a4a21e91ba920e69f9f6fc198773b5c/.github/workflows/warm-up-vercel.yml)
- https://github.com/blockprotocol/blockprotocol/pull/250

The idea is to fetch slow pages immediately after the deployment and then every several minutes. This reduces the probability of cold boots in the hands of end users.

I understand why a lambda fn would take 10+ seconds to load, e.g. if we need to establish a DB connection in a remote geographic region etc. However, if `getServerSideProps` just returns dummy data and Vercel still loads the page for the same amount of time, not sure what part of the app to blame. 🤔 Can the size of `node_modules` play a role here?

**Reply by Pieparker** [reply]
10s is remarkably... impractical! Have you considered trying to capture a slow request as a [HAR file](https://toolbox.googleapps.com/apps/har_analyzer/) to see _what_ is getting hung up? Are you on the latest version of Next.js?

The only other thing at play is `getServerSideProps` and the round-trip to the Lambda execution region itself. However, a HAR would tell us if we're losing 10s "in transit"!

**Reply by kachkaev** [reply]
We are using Next 12.1 and the app is deployed to Vercel via the standard GitHub integration. All pages are instant when running `yarn dev` or `yarn start` locally. TTFB is huge only when calling a ’cold’ lamda on Vercel (even if `getServerSideProps` has nothing in it).

Do you know how to generate a HAR file for a Vercel deployment? Any Vercel-specific debugging tips would be appreciated!

**Reply by Pieparker** [reply]
A HAR file is browser and request dependent, not Vercel dependent – [here's](https://www.inflectra.com/support/knowledgebase/kb254.aspx) how you can do it in Chrome or Firefox!

**Reply by kachkaev** [reply]
I see. Hmm not sure how this would help with the investigation because all I’m seeing in a browser is pretty much this: https://github.com/vercel/vercel/discussions/5422 (extra-long TTFB). It’s not clear why the lambda is taking so long, i.e. whether it is `getServerSideProps` itself or what happens on Vercel before it.

**Reply by Pieparker** [reply]
Got'cha! Well, we'd be happy to help further with a HAR file. The [very nature of a HAR file](https://www.keycdn.com/support/what-is-a-har-file) is "for identifying performance issues, such as bottlenecks and slow load times, and page rendering problems".


Feel free to create your own topic for us to work through if/when you have this, too! 🙏
